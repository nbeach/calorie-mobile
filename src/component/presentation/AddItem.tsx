import React from "react"
import {StyleSheet, Text, TextInput, TouchableOpacity, View} from "react-native"
import {border} from "../../util/StyleUtil"

export interface AddItemStateProps {
    readonly name: string
    readonly calories: number
}

export interface AddItemDispatchProps {
    readonly addItem: (name: string, calories: number) => void
}

export const AddItem = ({addItem, name, calories}: AddItemStateProps & AddItemDispatchProps) => {
    return <View style={styles.container}>
        <TextInput style={styles.item} placeholder={"Item"}/>
        <TextInput keyboardType="numeric" style={styles.calories} placeholder={"Cals"}/>

        <TouchableOpacity style={styles.button} onPress={() => addItem(name, calories)}>
            <Text style={styles.buttonText}>+</Text>
        </TouchableOpacity>
    </View>
}


const styles = StyleSheet.create({
    container: {
        flexDirection: "row",
        justifyContent: "space-between",
        padding: 10,
        backgroundColor: "#e4ebee",
    },
    item: {
        width: "60%",
        borderColor: "#AAAAAA",
        borderBottomWidth: 1,
        borderStyle: "solid",
        fontSize: 18,
    },
    calories: {
        width: "15%",
        borderColor: "#AAAAAA",
        borderBottomWidth: 1,
        borderStyle: "solid",
        fontSize: 18,
    },
    button: {
        width: "15%",
        fontSize: 18,
        alignItems: "center",
        backgroundColor: "#1b5fff",
        justifyContent: "center",
        ...border(1, "solid", "#EEEEEE"),
    },
    buttonText: {
        color: "#FFFFFF",
        fontSize: 18,
        fontWeight: "bold",
    },
})

