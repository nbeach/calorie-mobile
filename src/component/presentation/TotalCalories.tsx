import React from "react"
import {Text, View} from "react-native"

export interface TotalCaloriesStateProps {
    readonly totalCalories: number
}

export const TotalCalories = ({totalCalories}: TotalCaloriesStateProps) =>
    <View>
        <Text>{totalCalories.toString()}</Text>
    </View>
