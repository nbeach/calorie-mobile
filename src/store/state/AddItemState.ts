export interface AddItemState {
    readonly name: string
    readonly calories: string
}
