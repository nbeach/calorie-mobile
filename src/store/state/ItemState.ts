export interface ItemState {
    readonly id: string,
    readonly name: string
    readonly calories: number
}
