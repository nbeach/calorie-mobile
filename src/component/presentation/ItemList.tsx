import React from "react"
import {StyleSheet, Text, View} from "react-native"
import {ItemState} from "../../store/state/ItemState"
import {border} from "../../util/StyleUtil"

export interface ItemListProps {
    readonly items: ReadonlyArray<ItemState>
}

export const ItemList = (props: ItemListProps) => {
    return <View style={styles.container}>
        {props.items.map(item => <Text key={item.name} style={styles.item}>{item.name}</Text>)}
    </View>
}


const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: "center",
    },
    item: {
        padding: 10,
        width: "100%",
        textAlign: "center",
        fontSize: 18,
        ...border(1, "solid", "#AAAAAA"),
    },

})
