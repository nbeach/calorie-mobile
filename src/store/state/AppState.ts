import {ItemState} from "./ItemState"
import {AddItemState} from "./AddItemState"

export interface AppState {
    readonly items: ReadonlyArray<ItemState>,
    readonly addItem: AddItemState
}

