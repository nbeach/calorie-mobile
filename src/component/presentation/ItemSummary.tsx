import {ItemState} from "../../store/state/ItemState"
import {View} from "react-native"
import {TotalCalories} from "./TotalCalories"
import {ItemList} from "./ItemList"
import React from "react"

export interface ItemSummaryStateProps {
    readonly totalCalories: number,
    readonly items: ReadonlyArray<ItemState>
}

export interface ItemSummaryDispatchProps {
    readonly removeItem: (id: string) => void
}

export const ItemSummary = ({totalCalories, items, removeItem}: ItemSummaryStateProps & ItemSummaryDispatchProps) =>
    <View>
        <TotalCalories totalCalories={totalCalories}/>
        <ItemList items={items} removeItem={removeItem}/>
    </View>
