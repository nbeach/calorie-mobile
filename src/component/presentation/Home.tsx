import React from 'react';
import {ItemList} from "./ItemList";
import {StyleSheet, View} from 'react-native';
import {AppState} from "../../store/state/AppState";
import {AddItemContainer} from "../container/AddItemContainer";

export interface HomeProps extends AppState{

}

export const Home = (props: HomeProps) => {
    return <View style={styles.container}>
        <AddItemContainer></AddItemContainer>
        <ItemList items={props.items}/>
    </View>
};


const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'space-evenly',
        alignItems: 'stretch',
    },
});
