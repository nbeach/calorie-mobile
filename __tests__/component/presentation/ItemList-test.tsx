import React from "react"
import {ItemList} from "../../../src/component/presentation/ItemList"
import {ItemState} from "../../../src/store/state/ItemState"

import renderer from "react-test-renderer"

describe(ItemList.name, () => {

    it("displays items", () => {
        const items: ReadonlyArray<ItemState> = [
            {name: "chicken", calories: 100},
        ]

        const tree = renderer.create(<ItemList items={items}/>).toJSON()
        expect(tree).toMatchSnapshot()
    })

})

