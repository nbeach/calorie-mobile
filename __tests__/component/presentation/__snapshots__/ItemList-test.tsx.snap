// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList displays items 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
/>
`;
